coderd/httpmw/taskparam.go

```diff
@@ -23,35 +29,81 @@ func TaskParam(r *http.Request) database.Task {
 	return task
 }
 
-// ExtractTaskParam grabs a task from the "task" URL parameter by UUID.
+// ExtractTaskParam grabs a task from the "task" URL parameter.
+// It supports two lookup strategies:
+// 1. Task UUID (primary)
+// 2. Task name scoped to owner (secondary)
+//
+// This middleware depends on ExtractOrganizationMembersParam being in the chain
+// to provide the owner context for name-based lookups.
 func ExtractTaskParam(db database.Store) func(http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
 			ctx := r.Context()
-			taskID, parsed := ParseUUIDParam(rw, r, "task")
-			if !parsed {
+
+			// Get the task parameter value. We can't use ParseUUIDParam here because
+			// we need to support non-UUID values (task names) and
+			// attempt all lookup strategies.
+			taskParam := chi.URLParam(r, "task")
+			if taskParam == "" {
+				httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+					Message: "\"task\" must be provided.",
+				})
 				return
 			}
-			task, err := db.GetTaskByID(ctx, taskID)
+
+			// Get owner from OrganizationMembersParam middleware for name-based lookups
+			members := OrganizationMembersParam(r)
+			ownerID := members.UserID()
+
+			task, err := fetchTaskWithFallback(ctx, db, taskParam, ownerID)
 			if err != nil {
-				if httpapi.Is404Error(err) {
-					httpapi.ResourceNotFound(rw)
+				if !httpapi.Is404Error(err) {
+					httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+						Message: "Internal error fetching task.",
+						Detail:  err.Error(),
+					})
 					return
 				}
-				httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
-					Message: "Internal error fetching task.",
-					Detail:  err.Error(),
-				})
+				httpapi.ResourceNotFound(rw)
```

Comment:
Nit: I'd suggest reversing the `if` condition so that internal server error is the last resort after we have tried handling all known cases (currently just one, but more future proof).